db/hints/resource_manager.cc

```diff
@@ -65,15 +65,17 @@ size_t resource_manager::sending_queue_length() const {
 
 const std::chrono::seconds space_watchdog::_watchdog_period = std::chrono::seconds(1);
 
-space_watchdog::space_watchdog(shard_managers_set& managers, per_device_limits_map& per_device_limits_map)
+space_watchdog::space_watchdog(shard_managers_set& managers, per_device_limits_map& per_device_limits_map, seastar::named_semaphore& operation_lock)
     : _shard_managers(managers)
     , _per_device_limits_map(per_device_limits_map)
+    , _operation_lock(operation_lock)
 {}
 
 void space_watchdog::start() {
     _started = seastar::async([this] {
         while (!_as.abort_requested()) {
             try {
+                const auto units = get_units(_operation_lock, 1).get();
```

Comment:
Also unrelated and no action needed: better to put long running lambdas into named functions, so that if we have a stack trace with this code, it is attributed to the named function rather than start().